L1Trigger/TrackerTFP/src/State.cc

```diff
@@ -0,0 +1,139 @@
+#include "L1Trigger/TrackerTFP/interface/State.h"
+
+using namespace std;
+using namespace trackerDTC;
+
+namespace trackerTFP {
+
+  //
+  State::State(State* state) :
+    dataFormats_(state->dataFormats_),
+    setup_(state->setup_),
+    track_(state->track_),
+    trackId_(state->trackId_),
+    parent_(state->parent_),
+    stub_(state->stub_),
+    layerMap_(state->layerMap_),
+    hitPattern_(state->hitPattern_),
+    x0_(state->x0_),
+    x1_(state->x1_),
+    x2_(state->x2_),
+    x3_(state->x3_),
+    C00_(state->C00_),
+    C01_(state->C01_),
+    C11_(state->C11_),
+    C22_(state->C22_),
+    C23_(state->C23_),
+    C33_(state->C33_),
+    numSkippedLayers_(state->numSkippedLayers_),
+    numConsistentLayers_(state->numConsistentLayers_)
+  {}
+
+  // proto state constructor
+  State::State(const DataFormats* dataFormats, TrackKFin* track, int trackId) :
+    dataFormats_(dataFormats),
+    setup_(dataFormats->setup()),
+    track_(track),
+    trackId_(trackId),
+    parent_(nullptr),
+    stub_(nullptr),
+    layerMap_(setup_->numLayers()),
+    hitPattern_(0, setup_->numLayers()),
+    numSkippedLayers_(0),
+    numConsistentLayers_(0)
+  {
+    // initial track parameter residuals w.r.t. found track
+    x0_ = 0.;
+    x1_ = 0.;
+    x2_ = 0.;
+    x3_ = 0.;
+    // initial uncertainties
+    C00_  = pow(dataFormats_->base(Variable::inv2R, Process::sf), 2);
+    C11_  = pow(dataFormats_->base(Variable::phiT, Process::sf), 2);
+    C22_  = pow(dataFormats_->base(Variable::cot, Process::sf), 2);
+    C33_  = pow(dataFormats_->base(Variable::zT, Process::sf), 2);
+    C01_  = 0.;
+    C23_  = 0.;
+    // first stub
```

Comment:
Change comment to "first stub from first layer on input track with stubs"

Reply:
done